packages/invoices/src/OutboundShipment/api/hooks/document/useOutboundDelete.ts

```diff
@@ -26,6 +27,23 @@ export const useOutboundDelete = () => {
       .filter(Boolean) as OutboundRowFragment[],
   }));
 
+  const deleteData = () => {
+    mutate(selectedRows, {
+      onSuccess: () => queryClient.invalidateQueries(api.keys.base()),
+    });
+    const deletedMessage = t('messages.deleted-invoices', {
+      count: selectedRows.length,
+    });
+    const successSnack = success(deletedMessage);
+    successSnack();
+  };
+
+  const confirmDelete = useConfirmationModal({
+    onConfirm: deleteData,
+    message: t('messages.delete-shipment-warning'),
+    title: t('heading.are-you-sure'),
+  });
+
```

Comment:
Initially tried putting the confirmation call in the Toolbar component, which seems more appropriate. However then it would show the confirmation regardless of the additional delete checks (canDelete, none selected), which was not good.

Putting in here means splitting "deleteAction" into two functions, one pre-confirmation, and called on confirmation. That okay?

Reply:
yeah, I agree. Would be nice in the Toolbar. return a promise? handle resolve and reject in Toolbar?
pass in methods for `onSuccess` etc?